qemu/target-i386/seg_helper.c

```diff
@@ -939,6 +939,7 @@ void helper_syscall(CPUX86State *env, int next_eip_addend)
     CPUState *cs = CPU(x86_env_get_cpu(env));
 
     cs->exception_index = EXCP_SYSCALL;
+    env->exception_is_int = 1;
```

Comment:
i need to look at the code closer, but this worries me, as "exception_is_int" is also used elsewhere, so you set this here may trigger some unwanted behaviour.

Reply:
to be safe (and future-proof), is it better to create a new field in env for this?